### ISSUE TITLE
Formatting selected lines only?

### ISSUE THREAD

**Issue by grigorye** [ruiaureliano/X-SwiftFormat#10]
It looks really nice except for missing support for formatting only the selection. Are you working on that? Or you would accept PR targeting that (I have almost zero experience with developing Xcode extensions but it does look like more or less easy task)?

**Comment by ruiaureliano** [comment]
@grigorye just added this as a card to make in the next version :)

**Comment by grigorye** [comment]
👍 Thanks! Actually I gave it a try (adding another XCSourceEditorCommand for formatting selected lines) and got it almost working, however I could not figure out proper way to paste the formatted output when formatting indented lines - swift-format kind of stripes the indentation, but it makes sense to keep it... I could use indentation of the first line as the "reference point", but I'm not sure whether it's correct approach.

OTOH, just to let you know, some time ago I also did a quick hack around the same idea (integrate swift-format into Xcode), but using Automator workflow to format the selection. This problem of lost indentation does not exist there - at least Xcode maintains the level of indentation relatively indentation of the first line of the input, when Automator workflow "output replaces selected text". Update: it looks like Xcode maintaining the level indentation in case of Automator workflow depends on the indentation settings (it works for me with "Indent when typing" enabled and with "Re-indent on paste" disabled).

Here's my Automator-based solution for the reference:
https://github.com/grigorye/Swift-Format-For-Xcode
